contracts/tvl-staking/RibbonDualStakingRewards.sol

```diff
@@ -0,0 +1,359 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+// Inheritance
+import "../interfaces/IDualStakingRewards.sol";
+import "./RewardsDistributionRecipient.sol";
+import "../common/Pausable.sol";
+
+// https://docs.synthetix.io/contracts/source/contracts/stakingrewards
+contract DualStakingRewards is
+  IDualStakingRewards,
+  RewardsDistributionRecipient,
+  ReentrancyGuard,
+  Pausable
+{
+  using SafeMath for uint256;
+  using SafeCast for uint256;
+  using SafeERC20 for IERC20;
+
+  struct Rewards {
+    uint128 token0;
+    uint128 token1;
+  }
+
+  /* ========== STATE VARIABLES ========== */
+
+  IERC20 public immutable rewardsToken0;
+  IERC20 public immutable rewardsToken1;
+  IERC20 public immutable stakingToken;
+  Rewards public rewardRate;
+  Rewards public rewardPerTokenStored;
+  uint256 public periodFinish = 0;
```

Comment:
are we committing to rbn and avax emissions being the same duration and end at the same time?

otherwise should be:
```
Rewards public periodFinish;
Rewards public rewardsDuration;
Rewards public lastUpdateTime;
```